src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -199,70 +184,101 @@ private boolean canGoPlanting(@NotNull Field currentField, @NotNull BuildingFarm
 
         if (shouldTryToGetSeed)
         {
-            if (walkToBuilding())
+            final int slot = worker.findFirstSlotInInventoryWith(currentField.getSeed());
+            if (slot != -1)
             {
-                return false;
+                requestSeeds = false;
+            }
+            if (!walkToBuilding())
+            {
+                if (isInHut(new ItemStack(currentField.getSeed())))
+                {
+                    requestSeeds = false;
+                }
+                shouldTryToGetSeed = requestSeeds;
             }
-            isInHut(new ItemStack(currentField.getSeed()));
-            shouldTryToGetSeed = false;
         }
 
-        return true;
+        return !shouldTryToGetSeed;
     }
 
     /**
-     * Executes the hoeing of the field.
-     *
-     * @return the next state.
-     */
-    @NotNull
-    private AIState hoe()
+      * The main work cycle of the Famer.
+      * This checks each block, harvests, tills, and plants.
+      */
+    private AIState cycle()
     {
-        @Nullable BuildingFarmer buildingFarmer = getOwnBuilding();
+        @Nullable final BuildingFarmer buildingFarmer = getOwnBuilding();
 
-        if (buildingFarmer == null || buildingFarmer.getCurrentField() == null)
-        {
-            return AIState.PREPARING;
-        }
-        @Nullable Field field = getOwnBuilding().getCurrentField();
+        if (buildingFarmer == null || checkForHoe()) return AIState.PREPARING;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Missing curly brace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00121)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00122)